sim/src/tlv.rs

```diff
@@ -282,10 +279,11 @@ impl ManifestGen for TlvGen {
     fn make_tlv(self: Box<Self>) -> Vec<u8> {
         let mut protected_tlv: Vec<u8> = vec![];
 
-        if self.protect_size > 0 {
+        if !self.dependencies.is_empty() {
```

Comment:
Maybe using `if self.protect_size() > 0` to avoid depending on `dependencies` (no pun intented!) too much?

Reply:
Yeah, that's probably fine. I think we'll want to abstract it a little more when we add the next protected TLV thing.